src/generators/AbstractGenerator.ts

```diff
@@ -23,16 +26,16 @@ export const defaultGeneratorOptions: CommonGeneratorOptions = {
  * Abstract generator which must be implemented by each language
  */
 export abstract class AbstractGenerator<
-  Options extends CommonGeneratorOptions = CommonGeneratorOptions, 
-  RenderCompleteModelOptions = any> {
+  Options extends CommonGeneratorOptions, 
+  RenderCompleteModelOptions> {
   constructor(
     public readonly languageName: string,
     public readonly options: Options
   ) { }
 
-  public abstract render(model: MetaModel, inputModel: InputMetaModel): Promise<RenderOutput>;
-  public abstract renderCompleteModel(model: MetaModel, inputModel: InputMetaModel, options: RenderCompleteModelOptions): Promise<RenderOutput>;
-  public abstract constrainToMetaModel(model: MetaModel): ConstrainedMetaModel;
+  public abstract render(model: MetaModel, inputModel: InputMetaModel, dependencyManager?: AbstractDependencyManager): Promise<RenderOutput>;
+  public abstract renderCompleteModel(model: MetaModel, inputModel: InputMetaModel, options: Partial<RenderCompleteModelOptions>, dependencyManager?: AbstractDependencyManager): Promise<RenderOutput>;
+  public abstract constrainToMetaModel(model: MetaModel, dependencyManager: AbstractDependencyManager): ConstrainedMetaModel;
```

Comment:
If we have `AbstractDependencyManager` in the generator option, why we still need that in arguments?